smack-tcp/src/main/java/org/jivesoftware/smack/tcp/XMPPTCPConnection.java

```diff
@@ -855,9 +856,12 @@ else if (keyStoreType != null) {
         }
 
         final HostnameVerifier verifier = getConfiguration().getHostnameVerifier();
+        // convert Unicode domain into ASCII Compatible Encoding to match RFC3280 dNSname IA5String constraint
+        // see also: https://bugzilla.mozilla.org/show_bug.cgi?id=280839#c1
+        String ace_domain = DnsName.from(config.getXMPPServiceDomain()).toString();
```

Comment:
What if the configured XMPP service domain is not a valid DnsName? Also please use `DnsName.ace` in those cases, as readers unfamiliar with the `DnsName` API may not know that `toString()` returns the ACE encoded version of the DNS name.

Reply:
> What if the configured XMPP service domain is not a valid DnsName?

What do you have in mind? An IP address? Should we wrap the call in try / catch?

Reply:
> What do you have in mind? An IP address?

Not every XMPP Domainpart is a valid DNS name. It doesn't even has to involve IP addresses: You do not even need to look at IP addresses - a typical IPv4 address String will probably pass as valid DNS Name - but consider for example the different length restrictions: Simply speaking 255 for DNS names and 1024 for Domainparts.

> Should we wrap the call in try / catch?

Yes, definitely. With a sensible fallback.

Also note that I force pushed to the 4.3 branch in my personal Smack repo and thus another unrelated commit appears in this PR. I keep the right to force push to branches in my github repo of Smack. You may want to consider issuing your PRs based on the branches of the official Smack repository, which is guranteed to be always fast-forward.